kube-runtime/src/cache/mod.rs

```diff
@@ -25,6 +29,87 @@ where
     stream.inspect_ok(move |event| store.apply_watcher_event(event))
 }
 
+/// A simple reflector cache around a store and an owned watcher
+pub struct Cache<K>
+where
+    K: Clone + Resource + Send + Sync + 'static,
+    K::DynamicType: Eq + Hash,
+{
+    reader: Store<K>,
+    cache: BoxStream<'static, std::result::Result<watcher::Event<K>, watcher::Error>>,
+}
+
+impl<K> Cache<K>
+where
+    K: Clone + Resource + DeserializeOwned + Debug + Send + Sync + 'static,
+    K::DynamicType: Eq + Hash + Clone + Default,
+{
+    /// Create a Cache on a reflector on a type `K`
+    ///
+    /// Takes an [`Api`] object that determines how the `Cache` listens for changes to the `K`.
+    ///
+    /// The [`ListParams`] controls to the possible subset of objects of `K` that you want to cache.
+    /// For the full set of objects `K` in the given `Api` scope, you can use [`ListParams::default`].
+    #[must_use]
+    pub fn new(api: Api<K>, lp: ListParams) -> Self {
```

Comment:
Feels like this can just be moved into the `ìmpl` block below with a `fn`-level `where` clause.

Reply:
I'm not sure how to do that; the writer requires an explicit `dyntime`.